protoToHsGenTest/Unwrap.hs

```diff
@@ -0,0 +1,273 @@
+{-# LANGUAGE DeriveGeneric     #-}
```

Comment:
Did you mean to check in this generated file?

Reply:
Yes, this one. Want to makes sure the generated Haskell module is correct.

Reply:
I usually try to avoid including generated files in version control.  The reason why is similar in spirit to the Haskell idea of "make invalid states unrepresentable".  When one file in a repository is generated from another file within the same repository then there is the potential for the two files to get out of sync (for example: the original file is updated, but then people forget to regenerate the generated file from the original file).  Instead, I usually prefer to tests to create generated files "on the fly" as they are needed, to prevent this from happening